File: basic_problems_c++/lec14_ham_de_quy/quy_doi_nhi_phan/solution.py
def binary(a):
    if(n != 0 and a == 0): return None
    if((n == 0 and a == 0) or a == 1): 
        print(a % 2, end = "")
        return None
    binary(a // 2)
    print(a % 2, end = "")


n = int(input())
binary(n)

File: basic_problems_c++/lec14_ham_de_quy/quy_doi_nhi_phan/solution.cpp
#include <iostream>
using namespace std;

long long n;

void binary(long long a)
{
    if (n != 0 && a == 0) 
        return;
    if ((a == 0 && n == 0) || a == 1) 
        return void(cout << (a % 2LL));

    binary(a / 2LL);

    cout << (a % 2LL); //bit cuoi cung cua so n;
} 

int main() 
{
    cin >> n;
    binary(n);
    return 0;
}